sound/soc/sof/trace.c

```diff
@@ -13,6 +13,216 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+#define TRACE_FILTER_MAX_IPC_ELEMENTS_PER_ENTRY 4
+
+static int trace_filter_append_elem(int32_t key, int32_t value,
+				    struct sof_ipc_trace_filter_elem *elem_list,
+				    size_t capacity, size_t *counter)
+{
+	if (*counter >= capacity)
+		return -ENOMEM;
+
+	elem_list[*counter].key = key;
+	elem_list[*counter].value = value;
+	++*counter;
+
+	return 0;
+}
+
+static int trace_filter_parse_entry(struct snd_sof_dev *sdev, const char *line,
+				    struct sof_ipc_trace_filter_elem *elem,
+				    size_t capacity, size_t *counter)
+{
+	int len = strlen(line);
+	size_t cnt = *counter;
+	int log_level;
+	int uuid_id;
+	int pipe_id;
+	int comp_id;
+	int read;
+	int ret;
+
+	ret = sscanf(line, " %d %x %d %d %n", &log_level, &uuid_id, &pipe_id, &comp_id, &read);
+	if (ret != 4 || read != len) {
```

Comment:
TRACE_FILTER_MAX_IPC_ELEMENTS_PER_ENTRY instead of 4?
in fact this macro should be called TRACE_FILTER_IPC_ELEMENTS_PER_ENTRY instead because we really do expect 4 items isnt it?

Reply:
`TRACE_FILTER_MAX_IPC_ELEMENTS_PER_ENTRY` is renamed to `TRACE_FILTER_ELEMENTS_PER_ENTRY`, so it now defines logger entry length instead maximal IPC elements amount